### PR TITLE
docs/porting: Add patching an application section

### PR DESCRIPTION
**Pull request by StefanJum**
For some reason it seems like the `patching` part of the `porting` section was missed when adding content from [`USoC'21` - basic app porting](https://usoc21.unikraft.org/docs/sessions/08-basic-app-porting/#patching-the-application).

Signed-off-by: Stefan Jumarea <stefanjumarea02@gmail.com>

### CODE REVIEW

**Review by eduardvintila — APPROVED** [review]
The patching section is now visible. Thanks for this, @StefanJum.
Reviewed-by: Eduard Vintilă eduard.vintila47@gmail.com

**Comment by StefanJum** [comment]
This will need rebasing once #126 #125 and #124 are merged, I added the commits from there to check the linter output.

File: content/en/docs/develop/porting.md [hunk 1]
```diff
@@ -541,3 +535,102 @@ When this option is enabled, we can either:
    }
    ```
 
+### Patching the application
```
Comment (razvand) [inline content/en/docs/develop/porting.md:538]:
```suggestion
### Patching the Application
```

File: content/en/docs/develop/porting.md [hunk 2]
```diff
@@ -541,3 +535,102 @@ When this option is enabled, we can either:
...
+The process usually occurs during [steps 5 and 6 of providing build files](docs/develop/porting/#providing-build-files) of the application or library in question.
+During this process, we are expected to see compile-time and link-time errors from `gcc` as we add new files to the build and make fixes.
+
+The `iperf3` application port to Unikraft has four patches in order to make it work.
+Let's discuss them and what they mean.
+The next section discusses how to create one of these patches.
+
+1. [The first patch](https://github.com/lancs-net/lib-iperf3/blob/staging/patches/0001-Fix-duplicate-import-of-netinet-tcp.h.patch) comes from an error which is thrown when compiling the `iperf_api.c` source file.
+   This file is 3rd to be compiled from the list of complete source files.
+   In this file, we are receiving a duplicate import of `<netinent/tcp.h>`, simply removing this import fixes it, so the patch addresses this issue.
```
Comment (razvand) [inline content/en/docs/develop/porting.md:555]:
```suggestion
   In this file, we are receiving a duplicate import of `<netinet/tcp.h>`, simply removing this import fixes it, so the patch addresses this issue.
```

File: content/en/docs/develop/porting.md [hunk 3]
```diff
@@ -541,3 +535,102 @@ When this option is enabled, we can either:
...
+
+1. [The third patch](https://github.com/lancs-net/lib-iperf3/blob/staging/patches/0003-Set-the-temp-path-to-the-root.patch) arises from an assumption about the host environment and the difference between Linux user space and a unikernel.
+   With a traditional host OS, we have a filesystem populated with known paths, for example `/tmp`.
+   `iperf3` assumed this path exists, however, in the case of where no filesystem is provided to the unikernel during boot, which should be possible in some cases, the `iperf3` application would crash since `/tmp` does not exist beforehand.
+   The patch solves this by setting the temporary (ramfs) path to `/`.
+   An alternative solution is to make this path at boot.
+
+1. Finally, [the fourth patch](https://github.com/lancs-net/lib-iperf3/blob/staging/patches/0004-Disable-use-of-mmap-and-replace-with-mmalloc-and-fr.patch) is once again to do with missing functionality from Unikraft.
+   In this case, the syscalls [`mmap`](https://linux.die.net/man/2/mmap) and [`munmap`](https://linux.die.net/man/2/munmap) are missing.
+   In this case, `iperf3`, used `mmap` simply to statically allocate a region of memory.
```
Comment (razvand) [inline content/en/docs/develop/porting.md:571]:
```suggestion
   In this case, `iperf3` used `mmap` simply to statically allocate a region of memory.
```

File: content/en/docs/develop/porting.md [hunk 4]
```diff
@@ -541,3 +535,102 @@ When this option is enabled, we can either:
...
+The above patches represent example use cases where patches may be necessary to fix the application when bringing it to Unikraft.
+The possibilities presented in this tutorial are non-exhaustive, so take care.
+
+The next section discusses in detail how to create a patch for the target
+application or library.
+
+#### Preparing a Patch for the Application
+
+When a change is identified and is to be provided as a patch to the application or library during the compilation, it can be done using the procedure identified in this section.
+Note that providing patches are an unfortunate workaround to the inherent differences between Linux user space applications and libraries and unikernels.
```
Comment (razvand) [inline content/en/docs/develop/porting.md:585]:
```suggestion
Note that providing patches is an unfortunate workaround to the inherent differences between Linux user space applications and libraries and unikernels.
```

**Review by razvand — CHANGES_REQUESTED** [review]
I think the `$` sign should be kept in front of commands to make a difference between the command and its output.

Also, my suggestion is to update the linter to not require the language name for triple backtick environments. `bash` should only be used for shell scripts, not for commands; if used for commands, the highlight engine will also consider the command output - which shouldn't be highlighted.

**Review by razvand — APPROVED** [review]
All good. Thanks, @StefanJum 

Reviewed-by: Razvan Deaconescu <razvan.deaconescu@cs.pub.ro>
Approved-by: Razvan Deaconescu <razvan.deaconescu@cs.pub.ro>

**Pull request merged**